cfme/tests/openstack/cloud/test_networks.py

```diff
@@ -0,0 +1,225 @@
+"""Tests for Openstack cloud networks, subnets and routers"""
+
+import fauxfactory
+import pytest
+
+from cfme.cloud.provider.openstack import OpenStackProvider
+from cfme.utils import testgen
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.log import logger
+from cfme.utils.wait import wait_for
+
+
+pytest_generate_tests = testgen.generate([OpenStackProvider],
+                                         scope='module')
+
+pytestmark = [pytest.mark.usefixtures("setup_provider_modscope")]
+
+
+SUBNET_CIDR = '11.11.11.0/24'
+
+
+def delete_entity(entity):
+    # TODO: replace this with neutron client request
+    try:
+        if entity.exists:
+            entity.delete()
+    except Exception:
+        logger.warning('Exception during network entity deletion - skipping..')
+
+
+def create_network(appliance, provider, is_external):
+    collection = appliance.collections.cloud_networks
+    network = collection.create(name=fauxfactory.gen_alpha(),
+                                tenant=provider.get_yaml_data()['tenant'],
```

Comment:
I believe this should actually be pulled from `provider.data['provisioning'].get('tenant')` as that is the current yaml structure for the CFME QE team, and we're provisioning here.  I suggest using `get()` for tenant field so that it will be None if not in the yamls, and won't be filled in the form.  Don't use `get` (just use dictionary ['tenant'] syntax) if you want it to fail in that case.

Reply:
Set it to use tenant from provisioning data